src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,48 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import core.basesyntax.userinvalidexeption.UserInvalidExeption;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        return storageDao.add(user);
+    }
+
+    private void validateUser(User user) {
+        validateUserLogin(user.getLogin());
+        validateUserPassword(user.getPassword());
+        validateUserAge(user.getAge());
+    }
+
+    private void validateUserLogin(String login) {
+        if (login == null) {
+            throw new UserInvalidExeption("Please enter the login!");
+        }
+        if (login.length() < MINIMAL_LENGTH) {
+            throw new UserInvalidExeption("Login " + login + " must be at least 6 characters!");
+        }
+        if (storageDao.get(login) != null) {
+            throw new UserInvalidExeption("Login " + login + " already taking!");
+        }
+    }
+
+    private void validateUserPassword(String password) {
+        if (password == null || password.isEmpty()) {
+            throw new UserInvalidExeption("Please enter password!");
+        }
+        if (password.length() < MINIMAL_LENGTH) {
+            throw new UserInvalidExeption("Password must be at least 6 characters");
+        }
+    }
+
+    private void validateUserAge(int age) {
+        if (age <= 18) {
```

Comment:
Move all hardcoded values to constant fields and give them informative names.

Reply:
done